widgets/gui/qt_nidaq_worker.py

```diff
@@ -16,30 +17,37 @@ def __init__(self, parameters, view, channel, *args, **kwargs):
         self.channel = int(channel)
         self.args = args
         self.kwargs = kwargs
+        self.thread_active = True
 
         self.daq_card = None
 
     @pyqtSlot()
     def run(self):
         logging.info(f"NIDaq Instance launched")
-
-        self.daq_card = NIdaq(exposure=self.parameters[0],
-                              nb_timepoints=self.parameters[1],
-                              scan_step=self.parameters[2],
-                              stage_scan_range=self.parameters[3],
-                              vertical_pixels=self.parameters[4],
-                              num_samples=self.parameters[5],
-                              offset_view1=self.parameters[6],
-                              offset_view2=self.parameters[7],
-                              view1_galvo1=self.parameters[8],
-                              view1_galvo2=self.parameters[9],
-                              view2_galvo1=self.parameters[10],
-                              view2_galvo2=self.parameters[11],
-                              stripe_reduction_range=self.parameters[12],
-                              stripe_reduction_offset=self.parameters[13])
-
-        self.daq_card.select_view(1)
-        self.daq_card.select_channel_remove_stripes(488)
+        while True:
+            time.sleep(1)
+            logging.info(self.thread_active)
+            if not self.thread_active:
+                break
+
+        #self.daq_card = NIdaq(exposure=self.parameters[0],
+        #                      nb_timepoints=self.parameters[1],
+        #                      scan_step=self.parameters[2],
+        #                      stage_scan_range=self.parameters[3],
+        #                      vertical_pixels=self.parameters[4],
+        #                      num_samples=self.parameters[5],
+        #                      offset_view1=self.parameters[6],
+        #                      offset_view2=self.parameters[7],
+        #                      view1_galvo1=self.parameters[8],
+        #                      view1_galvo2=self.parameters[9],
+        #                      view2_galvo1=self.parameters[10],
+        #                      view2_galvo2=self.parameters[11],
+        #                      stripe_reduction_range=self.parameters[12],
+        #                      stripe_reduction_offset=self.parameters[13])
+        #
+        #self.daq_card.select_view(1)
+        #self.daq_card.select_channel_remove_stripes(488)
```

Comment:
do not do this please, we will test it on Bin's machine. Once you have something ready to test on Bin's machine let me know.

Reply:
Ok I will change it

Reply:
thank you @ljjh20 !